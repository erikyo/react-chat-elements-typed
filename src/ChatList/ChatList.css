.rce-container-clist {
    display: block;
    overflow-y: auto;
}
