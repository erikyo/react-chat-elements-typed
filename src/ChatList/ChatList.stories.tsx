import type { Meta, StoryObj } from "@storybook/react";

import ChatList from "./ChatList";
import { chatListArray } from "../stories/example/utils/MessageTypes";

const meta = {
	component: ChatList,
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default = {
	args: {
		dataSource: Array.from({ length: 10 }).map((_, i) => chatListArray()),
	},
};
