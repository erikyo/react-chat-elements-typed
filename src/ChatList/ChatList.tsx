import type { FC, SetStateAction } from "react";
import classNames from "classnames";
import "./ChatList.css";

import ChatItem from "../ChatItem/ChatItem";
import Loader from "../Loader/Loader";
import type { IChatListProps } from "../types";

const ChatList: FC<IChatListProps> = ({ dataSource, ...props }) => {
	return (
		<div className={classNames("rce-container-clist", props.className)}>
			{dataSource?.map((el, index) => (
				<ChatItem key={`chat-item-${index.toString()}`} {...el} />
			)) ?? <Loader />}
		</div>
	);
};

export default ChatList;
