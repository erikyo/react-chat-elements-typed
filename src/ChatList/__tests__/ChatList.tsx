import { describe, expect, it } from "vitest";
import React from "react";
import ChatList from "../ChatList";
import { render } from "@testing-library/react";

describe("Avatar component", () => {
	it("should render without issues", () => {
		const component = render(<ChatList />);
		expect(
			component.container.querySelectorAll(".rce-container-clist").length,
		).toBeTruthy();
		expect(component.container).toMatchSnapshot();
	});
});
