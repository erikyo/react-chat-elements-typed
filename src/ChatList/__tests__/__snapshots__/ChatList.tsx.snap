// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar component > should render without issues 1`] = `
<div>
  <div
    class="rce-container-clist"
  />
</div>
`;
